daemon/c_vol.c

```diff
@@ -683,16 +716,25 @@ c_vol_mount_image(c_vol_t *vol, const char *root, const mount_entry_t *mntent)
 	if (!img)
 		goto error;
 
+	TRACE("Mount entry type: %d", mount_entry_get_type(mntent));
+
 	switch (mount_entry_get_type(mntent)) {
 	case MOUNT_TYPE_SHARED:
+		mountflags |= MS_RDONLY;
+		verityfs = true;
+		break;
 	case MOUNT_TYPE_DEVICE:
-		mountflags |= MS_RDONLY; // add read-only flag for shared or device images types
+		mountflags |= MS_RDONLY;
 		break;
 	case MOUNT_TYPE_OVERLAY_RO:
 		mountflags |= MS_RDONLY; // add read-only flag for upper image
 		overlay = true;
 		break;
 	case MOUNT_TYPE_SHARED_RW:
+		overlay = true;
+		shiftids = true;
+		verityfs = true;
```

Comment:
just add verityfs = true and fall trough as before

Reply:
Done